### PR TITLE
Improve dmenu(u)mount scripts.

### PR DESCRIPTION
**Pull request by The-King-of-Toasters**
**TLDR**: I ran the scripts through shellcheck, improved awk commands and shortened variable names.
Also, `printf` is a very versatile command. Use it (printf(1), printf(3)).

I watched your latest video and I wanted to make some improvements. First, you should check your scripts with [shellcheck](https://github.com/koalaman/shellcheck) (its in the repos). POSIX `sh` has a small subset of bash features that we take for granted, so it's wise to check the scripts for max portability. Here's what the output said:
```sh
In dmenumount line 11:
[[ "$mountable" = "" ]] && exit 1
^-- SC2039: In POSIX sh, [[ ]] is undefined.

In dmenumount line 13:
[[ "$chosen" = "" ]] && exit 1
^-- SC2039: In POSIX sh, [[ ]] is undefined.

In dmenumount line 20:
[[ "$mountpoint" = "" ]] && exit 1
^-- SC2039: In POSIX sh, [[ ]] is undefined.

In dmenumount line 21:
if [[ ! -d "$mountpoint" ]]; then
   ^-- SC2039: In POSIX sh, [[ ]] is undefined.

In dmenumount line 22:
	mkdiryn=$(echo -e "No\nYes" | dmenu -i -p  "$mountpoint does not exist. Create it?")
                       ^-- SC2039: In POSIX sh, echo flags are undefined.
                             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".

In dmenumount line 23:
	[[ "$mkdiryn" = Yes ]] && sudo mkdir -p "$mountpoint"
        ^-- SC2039: In POSIX sh, [[ ]] is undefined.

In dmenumount line 25:
sudo mount $chosen $mountpoint && pgrep -x dunst && notify-send "$chosen mounted to $mountpoint."
           ^-- SC2086: Double quote to prevent globbing and word splitting.
                   ^-- SC2086: Double quote to prevent globbing and word splitting.

In dmenuumount line 9:
exclusionregex="\(/boot\|/home\|/\)$"
                ^-- SC1117: Backslash is literal in "\(". Prefer explicit escaping: "\\(".
                       ^-- SC1117: Backslash is literal in "\|". Prefer explicit escaping: "\\|".
                              ^-- SC1117: Backslash is literal in "\|". Prefer explicit escaping: "\\|".
                                 ^-- SC1117: Backslash is literal in "\)". Prefer explicit escaping: "\\)".

In dmenuumount line 11:
[[ "$drives" = "" ]] && exit
^-- SC2039: In POSIX sh, [[ ]] is undefined.

In dmenuumount line 13:
[[ "$chosen" = "" ]] && exit
^-- SC2039: In POSIX sh, [[ ]] is undefined.

In dmenuumount line 14:
sudo umount $chosen && pgrep -x dunst && notify-send "$chosen unmounted."
            ^-- SC2086: Double quote to prevent globbing and word splitting.
```
Another fix was the parsing of `lsblk`.  Insead of having two `grep` commands, you can specify the columns you want using `lsblk -o` and let `awk` do the parsing:

```sh
$ lsblk -lp | grep part | grep -v "t /" | awk '{print $1, "(" $4 ")"}' #old
/dev/sdb1 (128M)
/dev/sdc2 (12G)
/dev/sdd1 (232.3G)
/dev/sdd2 (460M)
/dev/sdd3 (100M)

$ COLS="name,type,size,mountpoint"
$ lsblk -rpo "$COLS" | awk '$2=="part"&&$4==""{printf "%s (%s)\n",$1,$3}'
/dev/sdb1 (128M)
/dev/sdd1 (232.3G)
/dev/sdd2 (460M)
/dev/sdd3 (100M)
```
This has the benefit of ignoring swap partitions, which is why `/dev/sdc2` is missing. The same can be done for the umount script:
```sh
$ exclusionregex="\(/boot\|/home\|/\)$"
$ lsblk -lp | grep "t /" | grep -v "$exclusionregex" | awk '{print $1, "(" $4 ")", "on", $7}'
/dev/sda1 (931.2G) on /mnt/OLDDRIVE
/dev/sdb2 (2.7T) on /mnt/SEAGATE

$ lsblk -rpo "$COLS" | awk '$2=="part"&&$4!~/boot|home|SWAP/&&length($4)>1{printf "%s (%s) on %s\n",$1,$3,$4}'
/dev/sda1 (931.2G) on /mnt/OLDDRIVE
/dev/sdb2 (2.7T) on /mnt/SEAGATE
```
Finally, I shortened the var names to ones more concise:
```
mountable -> drives
mountpoint -> mp
mkdiryn -> opt
```

### CODE REVIEW

**Comment by LukeSmithxyz** [comment]
Funny, I was actually tweaking with this myself and was going to add the column changes. I didn't think to exclude SWAP since I don't use a swap partion myself.

I'll merge some of these changes with my own additions, but I'm going to wait a day or two so people can easily see the scripts as they appeared in the video without getting confused ;-) (most people are going to get annoying looking back in git history).

**Comment by LukeSmithxyz** [comment]
Oh and I neglected to say that I had only accidentally marked this as `sh` and didn't think of it. Obviously with all the `[[`s and such, it's meant to be bash.
